rust_snuba/src/strategies/clickhouse.rs

```diff
@@ -0,0 +1,180 @@
+use crate::config::ClickhouseConfig;
+use crate::types::BytesInsertBatch;
+use reqwest::header::{HeaderMap, HeaderValue, ACCEPT_ENCODING, CONNECTION};
+use reqwest::{Client, Response};
+use rust_arroyo::processing::strategies::run_task_in_threads::{
+    RunTaskFunc, RunTaskInThreads, TaskRunner,
+};
+use rust_arroyo::processing::strategies::{
+    CommitRequest, InvalidMessage, ProcessingStrategy, SubmitError,
+};
+use rust_arroyo::types::Message;
+use std::time::Duration;
+
+struct ClickhouseWriter {
+    client: ClickhouseClient,
+    skip_write: bool,
+}
+
+impl ClickhouseWriter {
+    pub fn new(client: ClickhouseClient, skip_write: bool) -> Self {
+        ClickhouseWriter { client, skip_write }
+    }
+}
+
+impl TaskRunner<BytesInsertBatch, BytesInsertBatch> for ClickhouseWriter {
+    fn get_task(&self, message: Message<BytesInsertBatch>) -> RunTaskFunc<BytesInsertBatch> {
+        let skip_write = self.skip_write;
+        let client: ClickhouseClient = self.client.clone();
+        Box::pin(async move {
+            let len = message.payload().rows.len();
+            let mut data = vec![];
+            for row in message.payload().rows {
+                data.extend(row);
+                data.extend(b"\n");
+            }
+
+            if skip_write {
+                log::info!("skipping write of {} rows", len);
+                return Ok(message);
+            }
```

Comment:
This part is actually intentional. It's temporary code that skips the clickhouse write (since it's not prod ready) - but we want it's performance to be more similar to the case where it is actually is performing the write (and construct the data object even if it's not needed)